### PR TITLE
Add automatic indentation to semgrep CLI output

### PR DESCRIPTION
**Pull request by underyx**
This PR makes output formatting look a bit worse; other parts of #7251 adapt various output lines to rely on auto-indentation.

PR checklist:

- [ ] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [ ] Tests included or PR comment includes a reproducible test plan
- [ ] Documentation is up-to-date
- [ ] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [ ] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by underyx** [comment]
cc @aryx

File: cli/src/semgrep/console.py [hunk 1]
```diff
@@ -25,18 +27,81 @@ class Title:
...
+
+        return None
+
+    def print(self, *args: Any, deindent: int = 0, **kwargs: Any) -> None:
+        indent = 0
+
+        for arg in args:
+            title = self.extract_title(arg)
+            if title is not None:
+                self.active_title = title
```
Comment (brendongo) [inline cli/src/semgrep/console.py:93]: what does setting self.active_title do?

Comment (brendongo) [reply cli/src/semgrep/console.py:93]: Looks like this is handling printing titles but not sure how. Would it also be more correct to special case printing strings so console correctly handles any other future objects passed in (aside from Title class)

Comment (brendongo) [reply cli/src/semgrep/console.py:93]: Oh self.active_title is var we created nvm

File: cli/src/semgrep/console.py [hunk 2]
```diff
@@ -25,18 +27,81 @@ class Title:
...
+    @staticmethod
+    def extract_title(current: Optional[RenderableType]) -> Optional[Title]:
+        while isinstance(current, Title) or hasattr(current, "renderable"):
+            if isinstance(current, Title):
+                return current
+            current = getattr(current, "renderable", None)
+
+        return None
+
+    def print(self, *args: Any, deindent: int = 0, **kwargs: Any) -> None:
```
Comment (brendongo) [inline cli/src/semgrep/console.py:81]: Can you add quick comment explaining how you got this to work. We keep around reference to last title in active_title and figure out indentation level from active_title object

**Review by brendongo — APPROVED** [review]

File: cli/src/semgrep/console.py [hunk 3]
```diff
@@ -25,18 +27,81 @@ class Title:
...
+        else:
+            indent = self.auto_indent_size
+
+        indent = max(0, indent - deindent)
+
+        super().print(Padding.indent(Group(*args), indent), **kwargs)
+
+
+console = AutoIndentingConsole(highlighter=None)
+console.width = min(console.width, 120)
```
Comment (brendongo) [inline cli/src/semgrep/console.py]: Do you want these global?

**Pull request merged**